src/esmockModule.js

```diff
@@ -22,14 +22,27 @@ const asFileURL = p => fileurlre.test(p) ? p : url.pathToFileURL(p)
 const objProto = Object.getPrototypeOf({})
 const isPlainObj = o => Object.getPrototypeOf(o) === objProto
 
+// see https://github.com/iambumblehead/esmock/issues/234
+//
+// node v20.6.x returns invalid import.meta.resolve result "{}"
+const assertImportMetaResolveReturn = (url, id, parent) => {
+  if (typeof url === 'string' || url === null)
+    return url
+
+  throw esmockErr.errModuleIdUrlInvalid(id, parent)
+}
+
 // when import.meta.resolve fails to resolve windows paths, fallback resolvewith
 const resolve = isMetaResolve ?
   (import.meta.resolve.constructor.name === 'AsyncFunction'
     ? async (id, p) => import.meta.resolve(id, asFileURL(p))
+      .then(url => assertImportMetaResolveReturn(url, id, p))
```

Comment:
it's not required - behavior changed since 20.6.0, where import.meta.resolve is sync (== Function)

Reply:
ok removing...

Reply:
it is removed

Reply:
Thx! From my standpoint, 'AsyncFunction' branch is legacy now, so we shouldn't add some false expectation (about possible return value).